File: app/src/test/java/com/mgaetan89/showsrage/adapter/HistoriesAdapter_GetItemCountTest.java
package com.mgaetan89.showsrage.adapter;

import com.mgaetan89.showsrage.model.History;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;
import java.util.Collections;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(Parameterized.class)
public class HistoriesAdapter_GetItemCountTest {
	@Parameterized.Parameter(0)
	public List<History> histories;

	@Parameterized.Parameter(1)
	public int itemCount;

	private HistoriesAdapter adapter;

	@Before
	public void before() {
		this.adapter = new HistoriesAdapter(this.histories);
	}

	@Test
	public void getItemCount() {
		assertThat(this.adapter.getItemCount()).isEqualTo(this.itemCount);
	}

	@After
	public void after() {
		this.adapter = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{null, 0},
				{Collections.emptyList(), 0},
				{Collections.singletonList(new History()), 1},
				{Arrays.asList(new History(), new History(), new History()), 3},
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/adapter/HistoriesAdapter_GetItemCountTest.kt
package com.mgaetan89.showsrage.adapter

import com.mgaetan89.showsrage.model.History
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized

@RunWith(Parameterized::class)
class HistoriesAdapter_GetItemCountTest(val histories: List<History>, val itemCount: Int) {
    private lateinit var adapter: HistoriesAdapter

    @Before
    fun before() {
        this.adapter = HistoriesAdapter(this.histories)
    }

    @Test
    fun getItemCount() {
        assertThat(this.adapter.itemCount).isEqualTo(this.itemCount)
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            return listOf(
                    arrayOf(emptyList<Any>(), 0),
                    arrayOf(listOf(History()), 1),
                    arrayOf(listOf(History(), History(), History()), 3)
            )
        }
    }
}
